contracts/ProtocolReserve/ProtocolShareReserve.sol

```diff
@@ -164,12 +174,14 @@ contract ProtocolShareReserve is AccessControlledV8, ReentrancyGuardUpgradeable,
         //because prime relies on getUnreleasedFunds and its return value may change after config update
         _accrueAndReleaseFundsToPrime();
 
-        for (uint256 i = 0; i < configs.length;) {
+        _ensureMaxLoops(configs.length);
```

Comment:
I don't think we have to check this length here. If `configs.length` is too big the `for-loop` will fail, and it's ok. I don't think we have to check `distributionTargets.length` before the inner `for-loop`, because in that point it doesn't take into account the updates (the new items that will be inserted).

I think we should invoke `_ensureMaxLoops` with the new length of `distributionTargets` at the end of this function, after inserting new configs, reverting the transaction if there are too many items